net/speedtest/speedtest_server.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package speedtest
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+)
+
+// GetListener takes in a host and port as strings and creates and returns
+// a listener for that host port pair.
+func GetListener(host, port string) (*net.TCPListener, error) {
```

Comment:
Similarly to using net.Dial in the other file, using net.Listen here would make this function basically unnecessary - the caller could just pass a net.Listener to StartServer directly.